### ISSUE TITLE
Support for MongoCollection<Document>

### ISSUE THREAD

**Issue by philippe-tr** [mongojack/mongojack#123]
The current JacksonDBCollection.wrap method expects a DBCollection, but the mongodb driver 3+ encourages the use of MongoCollection<Document>.

Is support for MongoCollection<Document> planned? If yes, when can it be expected?

**Comment by ahmed-isam** [comment]
is this issue still open?

**Comment by Savir** [comment]
:+1:

**Comment by BenRomberg** [comment]
See also #105.

**Comment by mancvso** [comment]
The new async driver does not include the DB type, so it will not work.

**Comment by alexeyOnGitHub** [comment]
right, the current version does not support MongoDB v.3.

**Comment by ahmed-isam** [comment]
is there any plans for supporting MongoDB v3, I am more than happy to help in testing

**Comment by lukepalmer** [comment]
The maintainers seem to be quite busy unfortunately. v3 support is available in the snapshot and there's also a bunch of outstanding pull requests.

If necessary I can look at republishing a fork into maven, let me know if people would be interested in this.

**Comment by BenRomberg** [comment]
@lukepalmer: That would be amazing! Please let me know as soon as a somewhat stable version is ready, I'd be keen to try it out!

**Comment by WillBroadbentMSM** [comment]
I'd be interested in this too :+1:

**Comment by benmccann** [comment]
I published 2.6.0 which uses the latest mongo driver

**Comment by trueinsider** [comment]
@mancvso that's what I wrote for usage with async driver: https://gist.github.com/trueinsider/78a7abc8dc0d4ec5810a

I'm too looking forward when MongoJack will support async driver and I won't need this.

**Comment by mancvso** [comment]
@trueinsider thank you!

**Comment by ryber** [comment]
+!

**Comment by gdrte** [comment]
:+1:

**Comment by julienlavergne** [comment]
+1

Any support would be appreciated about this feature.

Any idea?

**Comment by leisim** [comment]
+1 I'd really need this...

**Comment by benmccann** [comment]
Pull requests welcome. Unfortunately all the maintainers are probably too busy at the moment to get to this, but if someone takes the time to put together a PR then I would review it

**Comment by sazzer** [comment]
Any news on this? I notice that the latest version - 2.6.1 - depends on the MongoDB Java Driver 3.2.1, but the actual code still depends on DBCollection instead of MongoCollection.

**Comment by benmccann** [comment]
No, as stated in my last comment, we're not working on this. PRs are welcome

**Comment by janhvh** [comment]
We are just starting with Mongo and MongoJack, and we have to start calling deprecated Mongo methods... Too bad :-1: :-(
I understand that developers are busy, and Mongo seems to have completely rewritten their driver... Still a shame that in a brand new project we have to use all deprecated methods...

**Comment by janhvh** [comment]
BTW, already over a year that Mongo has deprecated DB, DBCollection etc.in favor of MongoDatabase, MongoCollection, etc.

Any idea if / when MongoJack will support the new stuff?

**Comment by dozd** [comment]
I've created small library to support mapping through MongoCollection and Codecs introduced version 3. You could take a look - https://github.com/dozd/mongo-mapper.

Guys in mongo are implementing their own version of POJO codecs and will be probably available with next major release of their drivers.

**Comment by cfogrady** [comment]
PR: https://github.com/mongojack/mongojack/pull/164
